Pod/Classes/UISideMenuNavigationController.swift

```diff
@@ -147,17 +147,52 @@ open class UISideMenuNavigationController: UINavigationController {
         if SideMenuManager.menuAllowPopIfPossible {
             for subViewController in navigationController.viewControllers {
                 if type(of: subViewController) == type(of: viewController) {
+                    if SideMenuManager.menuTabMode {
+                        navigationController.popToViewController(subViewController, animated: false)
+                        CATransaction.commit()
+                        return
+                    }
                     navigationController.popToViewController(subViewController, animated: animated)
                     CATransaction.commit()
                     return
                 }
             }
         }
         
+        if SideMenuManager.menuPreserveViewOnPush {
+            var viewControllers = navigationController.viewControllers
+            let filtered = viewControllers.filter{$0.restorationIdentifier == viewController.restorationIdentifier}
+            if let preservedView = filtered.first {
+                viewControllers = viewControllers.filter() { $0 !== preservedView }
+                viewControllers.append(preservedView)
+                if SideMenuManager.menuTabMode {
+                    navigationController.setViewControllers(viewControllers, animated: false)
```

Comment:
In my opinion, the presence of a back button loses the expected meaning with this behavior. Since you're using a side menu to navigate between view controllers (reorganizing the back stack each time) then the user's expectation of what "back" does not take them through the order they've viewed the screens, just the ones most recently seen.

Last changes for you:
* Include `setAnimationsEnabled` from my other comment.
* For any view controller you're putting at the end of the stack, set it's `.navigationItem.hidesBackButton = true` (unless you strongly disagree)

I think I need to refactor all of these menu push variants into an enum instead of individual Bool values, especially since setting one prevents the behavior of another later in the method. I will do this when I find time, and incorporate your changes with them.